packages/core/src/utils.js

```diff
@@ -343,6 +347,32 @@ export async function withRetries(fn, { count, onRetry, signal, throwOn }) {
   }
 }
 
+export function redactSecrets(data) {
+  const filepath = path.resolve(url.fileURLToPath(import.meta.url), '../secretPatterns.yml');
+  const secretPatterns = YAML.parse(readFileSync(filepath, 'utf-8'));
+
+  if (Array.isArray(data)) {
+    // Process each item in the array
+    return data.map(item => redactSecrets(item));
+  } else if (typeof data === 'object' && data !== null) {
+    // Process each key-value pair in the object
+    data.message = redactSecrets(data.message);
+  }
+  if (typeof data === 'string') {
+    for (const pattern of secretPatterns.patterns) {
+      data = data.replace(new RegExp(pattern.pattern.regex, 'g'), '[REDACTED]');
+    }
+  }
+  return data;
+}
+
```

Comment:
do we know the performance for this?
Can we check with the debug logs in percy-web buildkite build

Reply:
We are not going to run on all CLI logs only CI error logs only. Although I have run in several time on 500KB file it is taking around 60ms